### DISCUSSION TITLE
deploying zksync

### DISCUSSION THREAD

**Discussion by gibhub01** [Cyfrin/foundry-full-course-cu#2419] [Q&A]
![image](https://github.com/user-attachments/assets/cd8347df-b3c2-4490-8d03-960ec9ac5231)

compiling and deploying in zksync is failing  please i need help

**Comment by EngrPips** [comment]
What workspace are you working in?

**Reply by gibhub01** [reply]
I created a new work space on remix

**Reply by EngrPips** [reply]
All right, so switch back to the `default-workspace` and ensure your contract is inside a contracts folder, then retry the process. The images below show the step-by-step process of how I successfully compile and deploy my contract to sepolia-zksync on remix.

<img width="1440" alt="Screenshot 2024-09-08 at 08 07 05" src="https://github.com/user-attachments/assets/e7b9c3a5-51e2-4067-a1b3-dfb408994bc6">
<img width="1440" alt="Screenshot 2024-09-08 at 08 08 30" src="https://github.com/user-attachments/assets/e6c5969f-f7fb-4d10-a059-7850a78bb179">
<img width="1440" alt="Screenshot 2024-09-08 at 08 09 01" src="https://github.com/user-attachments/assets/0837b9ce-e4f3-4c5b-a438-4d158c6386ee">
<img width="1440" alt="Screenshot 2024-09-08 at 08 09 35" src="https://github.com/user-attachments/assets/a6b10cb2-3a5a-4bef-bacb-b1f46ecdfd65">

**Reply by Ayikoandrew** [reply]
Thanks, I have been struggling with this since yesterday, but I just deployed my first contract, lol.

**Reply by EngrPips** [reply]
Awesome, Keep crushing it.

**Comment by Anshpatwal** [comment]
> ![image](https://private-user-images.githubusercontent.com/78292316/365378932-cd8347df-b3c2-4490-8d03-960ec9ac5231.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU3OTc2MjIsIm5iZiI6MTcyNTc5NzMyMiwicGF0aCI6Ii83ODI5MjMxNi8zNjUzNzg5MzItY2Q4MzQ3ZGYtYjNjMi00NDkwLThkMDMtOTYwZWM5YWM1MjMxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTA4VDEyMDg0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZjMWRhMTUxNGEyZWJjOTE0MmFhZmI2NzJkNzk5NmNiMDA1M2Y0MzAzMDM1MWYxZTQyYmIyNGM1OThjZTgxNGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.wD92R-x_9ZOXgq4-FnkpTupxcGPtZtQRsEfxIkfR2-U)
> 
> compiling and deploying in zksync is failing please i need help

Hey the same thing happend to me even though i had my file in contract folder, i just deleted all the folders and created new contract folder and then created a file in it and then it worked perfectly.

**Reply by EngrPips** [reply]
Interesting
